### ISSUE TITLE
Separate download folders for users

### ISSUE THREAD

**Issue by sachaw** [jfurrow/flood#460]
The ability have each users torrents separate is needed.

**Comment by jfurrow** [comment]
Yeah, this is a good idea. I'm still not sure how best to handle multiple users...

**Comment by sachaw** [comment]
I think it could be as simple as adding an user id to each torrent, then setting the basedir to /downloads/`username` or something like that, so when the user logs in it will display torrents that have there userid

**Comment by varyoo** [comment]
What's wrong with a Flood instance per user?

**Comment by dcousens** [comment]
@varyoo that would be a `rtorrent` *and* `flood` instance per user?

**Comment by varyoo** [comment]
Yes of course!

**Comment by sachaw** [comment]
Very heavy on resources, terrible idea.

**Comment by varyoo** [comment]
>Very heavy on resources, terrible idea.

Please be more specific sir.

**Comment by wopian** [comment]
Requiring a separate instance of rtorrent & flood for each user does not scale at all and wastes excessive amounts of memory (rtorrent uses 40mb and flood 60mb when *idle*).

Thus the lowest tier from DigitalOcean would only be able to physically allow 5 users (with 0 memory left to do anything). The tier can easily support 20-30 users with a single instance and wouldn't ever have to worry about memory usage

**Comment by varyoo** [comment]
@wopian how many torrents for those values?

**Comment by wopian** [comment]
0

**Comment by sachaw** [comment]
Yea, separate instances are not the the way to go about this, like I said earlier, assigning the id of the user who added the torrent to the torrent in a db then calling all torrents with the logged in users id and display them, then we can have a single instance of rtorrent and flood.

**Comment by jfurrow** [comment]
A long time ago I think I suggested that multi-users be handled by multiple instances of rtorrent. I agree this is a bad idea.

I think true multiuser support would be cool, but would be a huge amount of work, and it's not a big priority for me. If someone else wants to take it on, then let's come to an agreement about how it should be implemented first.

**Comment by jvacek** [comment]
(summary of #518) What about multiple rtorrent instances and one flood server that can have multiple users that can each just connect to the right socket?
This will allow separate settings, torrents, home folders, limits, labels, etc.

**Comment by jvacek** [comment]
The #549 PR is good, has anyone else tested it out yet? nudge nudge :)
